index.html

```diff
@@ -190,8 +192,16 @@ <h2>
     def buildProtocol(self, addr):
         return Echo()
 
-endpoints.serverFromString(reactor, "tcp:1234").listen(EchoFactory())
-reactor.run()
+async def echo_server(reactor):
+    await endpoints.serverFromString(reactor, "tcp:1234").listen(EchoFactory())
```

Comment:
I think this should show how to use the port 0 one and how to make it so cancellation will stop listening